internal/codespaces/rpc/invoker.go

```diff
@@ -235,6 +235,19 @@ func (i *invoker) StartSSHServerWithOptions(ctx context.Context, options StartSS
 	return port, response.User, nil
 }
 
+func listenTCP() (*net.TCPListener, error) {
+	addr, err := net.ResolveTCPAddr("tcp", "127.0.0.1:0")
+	if err != nil {
+		return nil, fmt.Errorf("failed to build tcp address: %w", err)
+	}
+	listener, err := net.ListenTCP("tcp", addr)
+	if err != nil {
+		return nil, fmt.Errorf("failed to listen to local port over tcp: %w", err)
+	}
+
+	return listener, nil
+}
```

Comment:
Just curious--can we reuse the `ListenTCP` method above instead of reimplementing it here? Am I maybe missing a cyclic dependency?

Reply:
Yeah, it is a cyclic dependency between `codespaces` and `rpc` packages. The loop is `(rpc) invoker.go -> (codespaces) codespaces.go` and `(codespaces) states.go -> (rpc) invoker.go`

Reply:
I'm considered making a whole third package just for `ListenTCP` but that felt like overkill :)